proj3_chess/chess/chess_pieces.py

```diff
@@ -0,0 +1,151 @@
+from board import Board
+from piece_properties import Player
+from abc import ABC, abstractmethod
+from coordinates import Move, Spot, MoveVector
+from typing import Iterable, Optional
+
+RIGHT = MoveVector(x=1, y=0)
+LEFT = MoveVector(x=-1, y=0)
+UP = MoveVector(x=0, y=1)
+DOWN = MoveVector(x=0, y=-1)
+UP_RIGHT = MoveVector(x=1, y=1)
+UP_LEFT = MoveVector(x=-1, y=1)
+DOWN_RIGHT = MoveVector(x=1, y=-1)
+DOWN_LEFT = MoveVector(x=-1, y=-1)
+
+
+class ChessPiece(ABC):
+    PIECE_NAME = ""
+
+    def __init__(self, player: Player) -> None:
+        self.player = player
+        self.direction = player.direction.value
+        self.symbol = player.symbol_transformation(self.PIECE_NAME)
+
+    def add_to_board(self, board: Board, spot: Spot) -> None:
+        self.board = board
+        if board.state[spot.x][spot.y]:
+            raise ValueError("spot is already taken")
+        board.state[spot.x][spot.y] = self
+
+    def is_legal_move(self, spot: Spot) -> bool:
+        if not self.board.is_spot_within_range(spot):
+            return False
+        spot_state = self.board.state[spot.x][spot.y]
+        return spot_state == None or spot_state.player != self.player
+
+    def does_capture_piece(self, spot: Spot) -> bool:
+        if not self.is_legal_move(spot):
+            raise ValueError("Spot is out of board range")
+        spot_state = self.board.state[spot.x][spot.y]
+        return spot_state is not None and spot_state.player != self.player
+
+    def get_spot(self) -> Spot:
```

Comment:
it sucks that you need to do this. with bigger boards this easily becomes an extremely slow method.
I see you went with the approach of "A piece does not have a position" but there are better ways to implement that.